search-processors/src/main/java/org/opensearch/searchpipelines/questionanswering/generative/llm/DefaultLlmImpl.java

```diff
@@ -126,12 +128,24 @@ protected Map<String, String> getInputParameters(ChatCompletionInput chatComplet
         return inputParameters;
     }
 
-    protected ChatCompletionOutput buildChatCompletionOutput(ModelProvider provider, Map<String, ?> dataAsMap) {
+    protected ChatCompletionOutput buildChatCompletionOutput(ModelProvider provider, Map<String, ?> dataAsMap, String llmResponseField) {
 
         List<Object> answers = null;
         List<String> errors = null;
 
-        if (provider == ModelProvider.OPENAI) {
+        if (llmResponseField != null) {
+            String response = (String) dataAsMap.get(llmResponseField);
+            if (response != null) {
+                answers = List.of(response);
+            } else {
+                Map error = (Map) dataAsMap.get("error");
+                if (error != null) {
+                    errors = List.of((String) error.get("message"));
+                } else {
+                    errors = List.of("Unknown error or response.");
```

Comment:
add test?

Reply:
I do cover lines 142 - 143.  I ran the coverage report locally.

Reply:
interesting, why codecov is not reporting that?

Reply:
oh ok, I was asking about the line no: 149

Reply:
I added coverage for both cases in the new diff.

Reply:
Addressed both cases in the new diff.